pkg/flare/archive_dca.go

```diff
@@ -176,3 +183,95 @@ func getDCAWorkloadList() ([]byte, error) {
 
 	return getWorkloadList(fmt.Sprintf("https://%v:%v/workload-list?verbose=true", ipcAddress, config.Datadog.GetInt("cluster_agent.cmd_port")))
 }
+
+// getAgentDaemonSet retrieves the DaemonSet manifest of the Agent
+func getAgentDaemonSet(fb flarehelpers.FlareBuilder) error {
+	// The Agent DaemonSet name is based on the Helm chart template and added to the Cluster Agent as an environment variable
+	var agentDaemonsetName string
+	var releaseNamespace string
+	var agentDaemonset []byte
+
+	cl, err := apiserver.GetAPIClient()
+	if err != nil {
+		return log.Errorf("Can't create client to query the API Server: %s", err)
+	}
+	agentDaemonsetName = os.Getenv(HELM_AGENT_DAEMONSET)
+	releaseNamespace = os.Getenv(HELM_CHART_RELEASE_NAMESPACE)
+	if agentDaemonsetName == "" || releaseNamespace == "" {
+		return log.Errorf("Can't collect the Agent Daemonset name and/or namespace from the environment variables %s and %v", HELM_AGENT_DAEMONSET, HELM_CHART_RELEASE_NAMESPACE)
+	}
```

Comment:
Should we still create the `agent-daemonset.yaml` file with those error in it ? This could ease troubleshooting.

Same for all the other new function.

Reply:
With `AddFileFromFunc` you mentioned, the `flare_creation.log` will contain the errors why the different files could not be collected, so it should be sufficient ?